firebase-query.html

```diff
@@ -287,10 +303,15 @@
         __onFirebaseChildAdded: function(snapshot) {
           var value = snapshot.val();
           var key = snapshot.key;
+          var leaf = typeof value !== 'object';
 
           this._log('Firebase child_added:', key, value);
 
-          value.$key = key;
+          if (leaf) {
+            value = {$key: key, $val: value};
```

Comment:
Make sure to account for this in `setStoredValue` above. A good strategy might be to check if the last fragment of the path is `$val` and then remove that fragment if it is before setting the changed value.